### PR TITLE
Co 2427 Add language to CRM.REQUEST and an Auto detection for the lang

### PR DESCRIPTION
**Pull request by nfluckiger**

### CODE REVIEW

**Comment by ecino** [comment]
You have a branch rebase problem. Please remove commit CO-2444 from this PR.

File: crm_request/views/request.xml [hunk 1]
```diff
@@ -98,6 +98,7 @@
                             <field name="user_id" context="{'default_groups_ref': ['base.group_user', 'base.group_partner_manager', 'base.group_sale_salesman_all_leads']}"/>
                             <field name="claim_type" string="Type" required="True" options="{'no_create': True}"/>
                             <field name="alias_id" groups="base.group_no_one"/>
+                            <field name="language">Language</field>
                         </group>
                         <group>
```
Comment (eicher31) [inline crm_request/views/request.xml:102]: Please move the field language here.

File: crm_request/views/request.xml [hunk 2]
```diff
@@ -98,6 +98,7 @@
                             <field name="user_id" context="{'default_groups_ref': ['base.group_user', 'base.group_partner_manager', 'base.group_sale_salesman_all_leads']}"/>
                             <field name="claim_type" string="Type" required="True" options="{'no_create': True}"/>
                             <field name="alias_id" groups="base.group_no_one"/>
+                            <field name="language">Language</field>
```
Comment (eicher31) [inline crm_request/views/request.xml]:
```suggestion
                            <field name="language"/>
```
Is enough

File: crm_request/models/request.py [hunk 3]
```diff
@@ -139,6 +153,10 @@ def message_new(self, msg, custom_values=None):
             }, options)
             if partner:
                 defaults['partner_id'] = partner.id
+                defaults['language'] = partner.lang
+
+        if 'language' not in defaults:
+            defaults['language'] = self.detect_lang(defaults['description'])
```
Comment (eicher31) [inline crm_request/models/request.py]: The key description does not exist anymore here. You can do that after the super:
`request = self.browse(id)`
`request.language = self.detect_lang(request.description)`
@ecino what do you think?

**Review by eicher31 — CHANGES_REQUESTED** [review]

File: crm_request/models/request.py (hunk 3 above)
Comment (ecino) [reply crm_request/models/request.py]: Yes, it looks good. Note that `super` is returning the `id` of the request.

**Review by eicher31 — APPROVED** [review]

**Review by ecino — APPROVED** [review]

**Pull request merged**